web_src/js/features/repo-graph.ts

```diff
@@ -1,92 +1,64 @@
-import {hideElem, showElem, type DOMEvent} from '../utils/dom.ts';
+import {toggleClass} from '../utils/dom.ts';
 import {GET} from '../modules/fetch.ts';
 import {fomanticQuery} from '../modules/fomantic/base.ts';
 
 export function initRepoGraphGit() {
   const graphContainer = document.querySelector<HTMLElement>('#git-graph-container');
   if (!graphContainer) return;
 
-  document.querySelector('#flow-color-monochrome')?.addEventListener('click', () => {
-    document.querySelector('#flow-color-monochrome').classList.add('active');
-    document.querySelector('#flow-color-colored')?.classList.remove('active');
-    graphContainer.classList.remove('colored');
-    graphContainer.classList.add('monochrome');
+  const elColorMonochrome = document.querySelector<HTMLElement>('#flow-color-monochrome');
+  const elColorColored = document.querySelector<HTMLElement>('#flow-color-colored');
+  const toggleColorMode = (mode: 'monochrome' | 'colored') => {
+    toggleClass(graphContainer, 'monochrome', mode === 'monochrome');
+    toggleClass(graphContainer, 'colored', mode === 'colored');
+
+    toggleClass(elColorMonochrome, 'active', mode === 'monochrome');
+    toggleClass(elColorColored, 'active', mode === 'colored');
+
     const params = new URLSearchParams(window.location.search);
-    params.set('mode', 'monochrome');
-    const queryString = params.toString();
-    if (queryString) {
-      window.history.replaceState({}, '', `?${queryString}`);
-    } else {
-      window.history.replaceState({}, '', window.location.pathname);
-    }
-    for (const link of document.querySelectorAll('.pagination a')) {
+    params.set('mode', mode);
+    window.history.replaceState({}, '', `?${params.toString()}`);
```

Comment:
```suggestion
    window.history.replaceState(null, '', `?${params.toString()}`);
```

Don't need a state object.